kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSClassDeclarationImpl.kt

```diff
@@ -125,7 +125,7 @@ class KSClassDeclarationImpl private constructor(internal val ktClassOrObjectSym
 
     override fun asType(typeArguments: List<KSTypeArgument>): KSType {
         if (typeArguments.isNotEmpty() && typeArguments.size != asStarProjectedType().arguments.size) {
-            return KSErrorType
+            return KSErrorType()
```

Comment:
In this (and similar) cases, where an error type is used to signal the incorrect API usage rather than to represent a truly missing/unresolved type, I've decided to not provide any additional hints. If that's not okay or if I've misinterpreted something, let's discuss the matter.